### PR TITLE
[harbour-validator] Allow python3dist(pillow). JB#55733

### PR DESCRIPTION
**Pull request by vigejolla**
Please note that even while we allow the Pillow library, we make no
quarantees of backwards compatibility. The application developers are
responsible for keeping their app compatible when we switch to new
major versions of Pillow.

### CODE REVIEW

File: allowed_requires.conf [hunk 1]
```diff
@@ -105,6 +105,7 @@ python3dist(zope-interface)
 python3dist(sortedcontainers)
 python3dist(toml)
 python3dist(twisted)
+python3dist(pillow)
```
Comment (pvuorela) [inline allowed_requires.conf:108]: Maybe we should have a comment here that we take the liberty to update the package?

**Comment by poetaster** [comment]
This is great utility for me in supporting Imageworks, but also for future work on Scribble and Videoworks that's in the pipeline. I've already been testing what was developed with 7.2 with the Jolla 8.x versions and it has been clear sailing!

**Comment by vigejolla** [comment]
> This is great utility for me in supporting Imageworks, but also for future work on Scribble and Videoworks that's in the pipeline. I've already been testing what was developed with 7.2 with the Jolla 8.x versions and it has been clear sailing!

We are also upgrading to 9.0.0 before this goes out. There are some incompatible changes, but not too many: https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#backwards-incompatible-changes

**Comment by poetaster** [comment]
This is great utility for me in supporting Imageworks, but also for future work on Scribble and Videoworks that's in the pipeline. I've already been testing what was developed with 7.2 with the Jolla 8.x versions and it has been clear sailing!

> > This is great utility for me in supporting Imageworks, but also for future work on Scribble and Videoworks that's in the pipeline. I've already been testing what was developed with 7.2 with the Jolla 8.x versions and it has been clear sailing!
> 
> We are also upgrading to 9.0.0 before this goes out. There are some incompatible changes, but not too many: https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#backwards-incompatible-changes

Looks minimal indeed. Will 9.0.0 be available from Jolla before the 4.4 release or should I try a local build?

**Comment by vigejolla** [comment]
> Looks minimal indeed. Will 9.0.0 be available from Jolla before the 4.4 release or should I try a local build?

At this stage I can't promise it to be even in 4.4.0, although there are good chances it will get in.

File: allowed_requires.conf [hunk 2]
```diff
@@ -105,19 +105,21 @@ python3dist(zope-interface)
 python3dist(sortedcontainers)
 python3dist(toml)
 python3dist(twisted)
+# While we allow Pillow, we make no quarantees of backwards compatibility between releases.
```
Comment (saukko) [inline allowed_requires.conf:108]: Worth adding to this comment "Supported since Sailfish 4.4.0" or so?

Comment (vigejolla) [reply allowed_requires.conf]: Added comment

File: allowed_requires.conf (hunk 1 above)
Comment (vigejolla) [reply allowed_requires.conf:110]: comment added, and also comment that this is supported since 4.5.0.

**Review by pvuorela — APPROVED** [review]

**Pull request merged**